emulator/main.cpp

```diff
@@ -90,17 +120,25 @@ int main(int argc, char* argv[]) {
     top->eval();
     timestamp++;
 
+    if (top->io_imem_store) {
+      printf("warn: IMEM store\n");
+      should_exit = true;
+      unknown_reason = false;
+      exit_in_n_cycles = 10;
+    }
```

Comment:
What I found was that bugs that wrote to IMEM would only yield weird behaviour/crash the program at a much later point in a program. I would often spend a lot of time backtracing, just to find the original cause was IMEM writes. So this is a pragmatic way to stop the program early, since IMEM writes typically are bugs either way. However, for a bootloader, this is not a bug.

This should be incorporated into the emulator in some better way, however. Possibly find a way to make the emulator aware of whether IMEM writes should be treated as bugs or not.